src/Features/LanguageServer/Microsoft.CodeAnalysis.LanguageServer/HostWorkspace/ProjectTelemetry/ProjectLoadTelemetryReporter.cs

```diff
@@ -0,0 +1,110 @@
+    /// so that we are able to compare data accurately.
+    /// See https://github.com/OmniSharp/omnisharp-roslyn/blob/master/src/OmniSharp.MSBuild/ProjectLoadListener.cs#L36
+    /// </summary>
+    public static async Task ReportProjectLoadTelemetryAsync(ImmutableArray<string> metadataReferences, OutputKind? outputKind, ProjectFileInfo projectFileInfo, ProjectToLoad projectToLoad, ILogger logger, CancellationToken cancellationToken)
+    {
+        try
+        {
+            // Matches O# behavior to not report this event if no references found.
+            if (!metadataReferences.Any())
+            {
+                return;
+            }
+
+            var projectId = GetProjectId(projectToLoad);
+            var sessionId = VsTfmAndFileExtHashingAlgorithm.HashInput(s_sessionId.ToString());
+            // We have 1 project per tfm, so we'll report a telemetry event for each tfm.
+            var targetFramework = projectFileInfo.TargetFramework ?? string.Empty;
+
+            var projectCapabilities = projectFileInfo.ProjectTelemetryMetadata.ProjectCapabilities;
+
+            var hashedReferences = GetHashedReferences(metadataReferences);
+            var fileCounts = GetUniqueHashedFileExtensionsAndCounts(projectFileInfo);
+            var isSdkStyleProject = projectFileInfo.ProjectTelemetryMetadata.IsSdkStyle;
+
+            var projectEvent = new ProjectLoadTelemetryEvent(
+                ProjectId: projectId.HashedValue,
+                SessionId: sessionId.HashedValue,
+                // Matches how O# reports output kind - default is 0 (ConsoleApplication)
+                OutputKind: (int?)outputKind ?? 0,
+                ProjectCapabilities: projectCapabilities.ToArray(),
+                TargetFrameworks: new string[] { targetFramework },
+                References: hashedReferences.Select(h => h.HashedValue).ToArray(),
+                FileExtensions: fileCounts.Keys.Select(k => k.HashedValue).ToArray(),
+                FileCounts: fileCounts.Values.ToArray(),
+                SdkStyleProject: isSdkStyleProject);
+
+            await ReportEventAsync(projectEvent, cancellationToken);
+        }
+        catch (Exception ex)
+        {
+            // Don't fail project loading because we failed to report telemetry.  Just log a warning and move on.
+            logger.LogWarning($"Failed to get project telemetry data: {ex.ToString()}");
+        }
+    }
+
+    private static async Task ReportEventAsync(ProjectLoadTelemetryEvent telemetryEvent, CancellationToken cancellationToken)
+    {
+        var instance = LanguageServerHost.Instance;
+        Contract.ThrowIfNull(instance, nameof(instance));
+        var clientLanguageServerManager = instance.GetRequiredLspService<IClientLanguageServerManager>();
+        await clientLanguageServerManager.SendNotificationAsync("workspace/projectConfiguration", telemetryEvent, cancellationToken);
```

Comment:
```suggestion
        await clientLanguageServerManager.SendNotificationAsync("workspace/projectConfigurationTelemetry", telemetryEvent, cancellationToken);
```

or something? I can imagine us using a similar name for if we create a configuration dropdown or something.